### ISSUE TITLE
PYTHONPATH set by spack is broken on Fedora 37

### ISSUE THREAD

**Issue by balay** [LLNL/spack#33923]
### Steps to reproduce

- setup system python in ~/.spack/packages.yaml
```
  python:
    buildable: False
    externals:
    - spec: python@3.11.0%gcc@12.2.1
      prefix: /usr
```
- spack install py-numpy

### Error message

```console
balay@pj01 spack.orig]$ eval `./bin/spack load --sh   py-numpy`
[balay@pj01 spack.orig]$ python3 -c "import numpy; print(numpy.__file__)"
/usr/lib64/python3.11/site-packages/numpy/__init__.py
[balay@pj01 spack.orig]$ ./bin/spack find -p py-numpy
-- linux-fedora37-skylake / gcc@12.2.1 --------------------------
py-numpy@1.23.4  /home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-numpy-1.23.4-27kw7ucbqregxz6khsa3jxenjhelzbqh
==> 1 installed package
[balay@pj01 spack.orig]$ echo $PYTHONPATH
/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-setuptools-59.4.0-rgac3zhrekwrzs4w4hcs3omsutc6n5cv/local/lib64/python3.11/site-packages:/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-setuptools-59.4.0-rgac3zhrekwrzs4w4hcs3omsutc6n5cv/local/lib/python3.11/site-packages:/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-numpy-1.23.4-27kw7ucbqregxz6khsa3jxenjhelzbqh/local/lib64/python3.11/site-packages:/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-numpy-1.23.4-27kw7ucbqregxz6khsa3jxenjhelzbqh/local/lib/python3.11/site-packages:/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-setuptools-59.4.0-rgac3zhrekwrzs4w4hcs3omsutc6n5cv/local/lib64/python3.11/site-packages:/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-setuptools-59.4.0-rgac3zhrekwrzs4w4hcs3omsutc6n5cv/local/lib/python3.11/site-packages:/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-cython-0.29.32-zhikpenzqk6klv425hhfkil2is7zvla3/local/lib64/python3.11/site-packages:/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-cython-0.29.32-zhikpenzqk6klv425hhfkil2is7zvla3/local/lib/python3.11/site-packages:/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-wheel-0.37.1-jra3r6z6soemlggorr7tgz2pzms5lsl4/local/lib64/python3.11/site-packages:/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-wheel-0.37.1-jra3r6z6soemlggorr7tgz2pzms5lsl4/local/lib/python3.11/site-packages:/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-setuptools-59.4.0-rgac3zhrekwrzs4w4hcs3omsutc6n5cv/local/lib64/python3.11/site-packages:/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-setuptools-59.4.0-rgac3zhrekwrzs4w4hcs3omsutc6n5cv/local/lib/python3.11/site-packages:/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-pip-22.2.2-gx5wdlrcl2pai3pvhucy4yw42g2rst6i/local/lib64/python3.11/site-packages:/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-pip-22.2.2-gx5wdlrcl2pai3pvhucy4yw42g2rst6i/local/lib/python3.11/site-packages
[balay@pj01 spack.orig]$ find /home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-numpy-1.23.4-27kw7ucbqregxz6khsa3jxenjhelzbqh -name *.so
/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-numpy-1.23.4-27kw7ucbqregxz6khsa3jxenjhelzbqh/lib64/python3.11/site-packages/numpy/core/_multiarray_tests.cpython-311-x86_64-linux-gnu.so
/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-numpy-1.23.4-27kw7ucbqregxz6khsa3jxenjhelzbqh/lib64/python3.11/site-packages/numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so
/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-numpy-1.23.4-27kw7ucbqregxz6khsa3jxenjhelzbqh/lib64/python3.11/site-packages/numpy/core/_operand_flag_tests.cpython-311-x86_64-linux-gnu.so
/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-numpy-1.23.4-27kw7ucbqregxz6khsa3jxenjhelzbqh/lib64/python3.11/site-packages/numpy/core/_rational_tests.cpython-311-x86_64-linux-gnu.so
/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-numpy-1.23.4-27kw7ucbqregxz6khsa3jxenjhelzbqh/lib64/python3.11/site-packages/numpy/core/_simd.cpython-311-x86_64-linux-gnu.so
/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-numpy-1.23.4-27kw7ucbqregxz6khsa3jxenjhelzbqh/lib64/python3.11/site-packages/numpy/core/_struct_ufunc_tests.cpython-311-x86_64-linux-gnu.so
/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-numpy-1.23.4-27kw7ucbqregxz6khsa3jxenjhelzbqh/lib64/python3.11/site-packages/numpy/core/_umath_tests.cpython-311-x86_64-linux-gnu.so
/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-numpy-1.23.4-27kw7ucbqregxz6khsa3jxenjhelzbqh/lib64/python3.11/site-packages/numpy/fft/_pocketfft_internal.cpython-311-x86_64-linux-gnu.so
/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-numpy-1.23.4-27kw7ucbqregxz6khsa3jxenjhelzbqh/lib64/python3.11/site-packages/numpy/linalg/_umath_linalg.cpython-311-x86_64-linux-gnu.so
/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-numpy-1.23.4-27kw7ucbqregxz6khsa3jxenjhelzbqh/lib64/python3.11/site-packages/numpy/linalg/lapack_lite.cpython-311-x86_64-linux-gnu.so
/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-numpy-1.23.4-27kw7ucbqregxz6khsa3jxenjhelzbqh/lib64/python3.11/site-packages/numpy/random/_bounded_integers.cpython-311-x86_64-linux-gnu.so
/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-numpy-1.23.4-27kw7ucbqregxz6khsa3jxenjhelzbqh/lib64/python3.11/site-packages/numpy/random/_common.cpython-311-x86_64-linux-gnu.so
/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-numpy-1.23.4-27kw7ucbqregxz6khsa3jxenjhelzbqh/lib64/python3.11/site-packages/numpy/random/_generator.cpython-311-x86_64-linux-gnu.so
/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-numpy-1.23.4-27kw7ucbqregxz6khsa3jxenjhelzbqh/lib64/python3.11/site-packages/numpy/random/_mt19937.cpython-311-x86_64-linux-gnu.so
/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-numpy-1.23.4-27kw7ucbqregxz6khsa3jxenjhelzbqh/lib64/python3.11/site-packages/numpy/random/_pcg64.cpython-311-x86_64-linux-gnu.so
/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-numpy-1.23.4-27kw7ucbqregxz6khsa3jxenjhelzbqh/lib64/python3.11/site-packages/numpy/random/_philox.cpython-311-x86_64-linux-gnu.so
/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-numpy-1.23.4-27kw7ucbqregxz6khsa3jxenjhelzbqh/lib64/python3.11/site-packages/numpy/random/_sfc64.cpython-311-x86_64-linux-gnu.so
/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-numpy-1.23.4-27kw7ucbqregxz6khsa3jxenjhelzbqh/lib64/python3.11/site-packages/numpy/random/bit_generator.cpython-311-x86_64-linux-gnu.so
/home/balay/spack.orig/opt/spack/linux-fedora37-skylake/gcc-12.2.1/py-numpy-1.23.4-27kw7ucbqregxz6khsa3jxenjhelzbqh/lib64/python3.11/site-packages/numpy/random/mtrand.cpython-311-x86_64-linux-gnu.so
```

### Information on your system

```console
[balay@pj01 spack.orig]$ ./bin/spack debug report
* **Spack:** 0.20.0.dev0 (0811f81a09bbe87274a9aae093245ff7d8a4af9c)
* **Python:** 3.11.0
* **Platform:** linux-fedora37-skylake
* **Concretizer:** clingo
```

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have run the failing commands in debug mode and reported the output

**Comment by balay** [comment]
Here `PYTHONPATH` is incorrectly set to `pkg-dir/local/lib64/python3.11/site-packages`  [and pkg-dir/local/lib/...] instead of the correct `pkg-dir/lib64/python3.11/site-packages`?

I don't see this issue on ubuntu-20.04 - where PYTHONPATH is set to `pkg-dir/lib/python3.8/site-packages` - and the python module is installed correctly in the same location

```console
balay@pj02:~/spack$ eval `./bin/spack load --sh   py-numpy`
balay@pj02:~/spack$ python3 -c "import numpy; print(numpy.__file__)"
/home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-numpy-1.23.4-cq6nirghx53zmorispwgeybfuflbfbkk/lib/python3.8/site-packages/numpy/__init__.py
balay@pj02:~/spack$ ./bin/spack find -p py-numpy
-- linux-ubuntu20.04-skylake / gcc@9.4.0 ------------------------
py-numpy@1.23.4  /home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-numpy-1.23.4-cq6nirghx53zmorispwgeybfuflbfbkk
==> 1 installed package
balay@pj02:~/spack$ echo $PYTHONPATH
/home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-setuptools-59.4.0-y63dqs7o4wa3i7zsnlk2hzwsfasrbw5i/lib/python3.8/site-packages:/home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-numpy-1.23.4-cq6nirghx53zmorispwgeybfuflbfbkk/lib/python3.8/site-packages:/home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-setuptools-59.4.0-y63dqs7o4wa3i7zsnlk2hzwsfasrbw5i/lib/python3.8/site-packages:/home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-cython-0.29.32-wzc2nnlguxayfujd5mjrh7n6v5lrfo3y/lib/python3.8/site-packages:/home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-wheel-0.37.1-jvrestxubjmdjlm3eouwevehcwurw2th/lib/python3.8/site-packages:/home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-setuptools-59.4.0-y63dqs7o4wa3i7zsnlk2hzwsfasrbw5i/lib/python3.8/site-packages:/home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-pip-22.2.2-467ws6wktewonn4h4tv5brnaqv3pdqxf/lib/python3.8/site-packages
balay@pj02:~/spack$ find /home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-numpy-1.23.4-cq6nirghx53zmorispwgeybfuflbfbkk -name *.so
/home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-numpy-1.23.4-cq6nirghx53zmorispwgeybfuflbfbkk/lib/python3.8/site-packages/numpy/fft/_pocketfft_internal.cpython-38-x86_64-linux-gnu.so
/home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-numpy-1.23.4-cq6nirghx53zmorispwgeybfuflbfbkk/lib/python3.8/site-packages/numpy/linalg/_umath_linalg.cpython-38-x86_64-linux-gnu.so
/home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-numpy-1.23.4-cq6nirghx53zmorispwgeybfuflbfbkk/lib/python3.8/site-packages/numpy/linalg/lapack_lite.cpython-38-x86_64-linux-gnu.so
/home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-numpy-1.23.4-cq6nirghx53zmorispwgeybfuflbfbkk/lib/python3.8/site-packages/numpy/random/bit_generator.cpython-38-x86_64-linux-gnu.so
/home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-numpy-1.23.4-cq6nirghx53zmorispwgeybfuflbfbkk/lib/python3.8/site-packages/numpy/random/_bounded_integers.cpython-38-x86_64-linux-gnu.so
/home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-numpy-1.23.4-cq6nirghx53zmorispwgeybfuflbfbkk/lib/python3.8/site-packages/numpy/random/_mt19937.cpython-38-x86_64-linux-gnu.so
/home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-numpy-1.23.4-cq6nirghx53zmorispwgeybfuflbfbkk/lib/python3.8/site-packages/numpy/random/_philox.cpython-38-x86_64-linux-gnu.so
/home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-numpy-1.23.4-cq6nirghx53zmorispwgeybfuflbfbkk/lib/python3.8/site-packages/numpy/random/mtrand.cpython-38-x86_64-linux-gnu.so
/home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-numpy-1.23.4-cq6nirghx53zmorispwgeybfuflbfbkk/lib/python3.8/site-packages/numpy/random/_sfc64.cpython-38-x86_64-linux-gnu.so
/home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-numpy-1.23.4-cq6nirghx53zmorispwgeybfuflbfbkk/lib/python3.8/site-packages/numpy/random/_pcg64.cpython-38-x86_64-linux-gnu.so
/home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-numpy-1.23.4-cq6nirghx53zmorispwgeybfuflbfbkk/lib/python3.8/site-packages/numpy/random/_common.cpython-38-x86_64-linux-gnu.so
/home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-numpy-1.23.4-cq6nirghx53zmorispwgeybfuflbfbkk/lib/python3.8/site-packages/numpy/random/_generator.cpython-38-x86_64-linux-gnu.so
/home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-numpy-1.23.4-cq6nirghx53zmorispwgeybfuflbfbkk/lib/python3.8/site-packages/numpy/core/_multiarray_umath.cpython-38-x86_64-linux-gnu.so
/home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-numpy-1.23.4-cq6nirghx53zmorispwgeybfuflbfbkk/lib/python3.8/site-packages/numpy/core/_simd.cpython-38-x86_64-linux-gnu.so
/home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-numpy-1.23.4-cq6nirghx53zmorispwgeybfuflbfbkk/lib/python3.8/site-packages/numpy/core/_operand_flag_tests.cpython-38-x86_64-linux-gnu.so
/home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-numpy-1.23.4-cq6nirghx53zmorispwgeybfuflbfbkk/lib/python3.8/site-packages/numpy/core/_multiarray_tests.cpython-38-x86_64-linux-gnu.so
/home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-numpy-1.23.4-cq6nirghx53zmorispwgeybfuflbfbkk/lib/python3.8/site-packages/numpy/core/_struct_ufunc_tests.cpython-38-x86_64-linux-gnu.so
/home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-numpy-1.23.4-cq6nirghx53zmorispwgeybfuflbfbkk/lib/python3.8/site-packages/numpy/core/_umath_tests.cpython-38-x86_64-linux-gnu.so
/home/balay/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/py-numpy-1.23.4-cq6nirghx53zmorispwgeybfuflbfbkk/lib/python3.8/site-packages/numpy/core/_rational_tests.cpython-38-x86_64-linux-gnu.so
```

**Comment by balay** [comment]
Note: This issue triggers a build failure of `py-slepc4py` - as it fails to find/load dependent pkg `py-petsc4py`

```
    File "/home/balay/spack.orig/spack-stage/spack-stage-py-slepc4py-3.18.1-cckbtk2paytwmshljmbrdcn6xpfdk7ap/spack-src/setup.py", line 282, in build_sources
      import petsc4py
  ModuleNotFoundError: No module named 'petsc4py'
```

**Comment by adamjstewart** [comment]
This is a known issue. For more discussion, see:

* https://github.com/pypa/pip/issues/10978
* https://github.com/spack/spack/pull/31939
* https://github.com/spack/spack/issues/32662
* https://github.com/spack/spack/issues/32769

TL;DR: the system Python on the majority of Linux distros (not just Fedora) contains a number of ungodly hacks to change the default behavior of distutils and setuptools. These is no way to undo these hacks, so all packages will be installed to `<prefix>/local` instead of `<prefix>`, or `<prefix>/lib64` instead of `<prefix>/lib`. This breaks all auto-detect behavior for `PATH` and `PYTHONPATH`. I would strongly recommend not using external Python on any Linux system and instead letting Spack install Python. Or complaining to the maintainers of RHEL/Debian and asking them to stop patching distutils/setuptools so that they don't change the default behavior of these libraries.

@pradyunsg correct me if I'm wrong on any of the above.

**Comment by adamjstewart** [comment]
Closing as a duplicate of #32662
